### PR TITLE
Master remove hardcoded names of clients

### PR DESCRIPTION
**Pull request by Bischoff**
... and generalize the use of DEFAULT_TIMEOUT

(WIP: should work, but not tested yet)

### CODE REVIEW

**Comment by mantel** [comment]
Looks very good to me!

File: features/step_definitions/salt_steps.rb [hunk 1]
```diff
@@ -756,8 +756,9 @@
 
 Given(/^the salt-master can reach "([^"]*)"$/) do |minion|
   begin
+    # 300 is the default 1st keepalive interval for the minion
     # where it realizes the connection is stuck
-    Timeout.timeout(DEFAULT_TIMEOUT) do
+    Timeout.timeout(300) do
```
Comment (MalloZup) [inline features/step_definitions/salt_steps.rb]: i would use here like 

```
  KEEP_ALIVE_TIMEOUT= 300
  Timeout.timeout(KEEP_ALIVE_TIMEOUT) do
```

File: features/step_definitions/salt_steps.rb [hunk 2]
```diff
@@ -25,7 +25,7 @@
     start = Time.now
     # 300 is the default 1st keepalive interval for the minion
     # where it realizes the connection is stuck
-    Timeout.timeout(DEFAULT_TIMEOUT) do
+    Timeout.timeout(300) do
```
Comment (MalloZup) [inline features/step_definitions/salt_steps.rb]: Can you introduce a var here?

Otherwise it seems a Magic number

Comment (MalloZup) [inline features/step_definitions/salt_steps.rb]: a var like KEEP_ALIVE_TIMEOUT

File: features/step_definitions/navigation_steps.rb [hunk 3]
```diff
@@ -34,6 +34,24 @@
     raise "Couldn't find the #{text} in webpage"
   end
 end
+
+When(/^I wait until I see the name of "([^"]*)", refreshing the page$/) do |target|
+  case target
```
Comment (MalloZup) [inline features/step_definitions/navigation_steps.rb]: here,  we see that we should get rid of the var $minion_fullhostname and make it as an attribute of the object.

THen we could use 
```get_node(target) ```
```  step %(I wait until i see "#{node.fullhostname}" text, refreshing the page```

**Comment by Bischoff** [comment]
Very nice idea, the variable (from a ruby pov that's a constant, because it's upper case)

We already have this comment (don't know who did it), but having the variable too does not harm.

**Comment by Bischoff** [comment]
Ouchie, it looks like i'm missing a call to get_node, nice catch dario

**Comment by Bischoff** [comment]
> here, we see that we should get rid of the var $minion_fullhostname and make it as an attribute of the object.

I examined this - you're right, but it's a big change that affect many places. So I will not do it now. We need to put it on the refactoring TODO and do it correctly in a different pull request.

**Comment by MalloZup** [comment]
@Bischoff sure, that's why i opened an issue, because for this PR is to much.

**Pull request merged**